azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -257,16 +267,58 @@ private void addNetworkInterfaceCard() {
               .type("Microsoft.Network/networkInterfaces")
               .location(location)
               .apiVersion(STORAGE_API_VERSION)
-              .dependsOn(Arrays.asList("[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]",
-                      "[concat('Microsoft.Network/virtualNetworks/', variables('virtualNetworkName'))]"))
+              .dependsOn(depends)
               .properties(networkInterfaceCardProperties)
               .build();
 
       resources.add(networkInterfaceCard);
    }
 
-   private void addVirtualMachine() {
+   private void addNetworkSecurityGroup() {
+      int ports[] = options.getInboundPorts();
+      if ((ports != null) && (ports.length > 0)) {
+         variables.put("networkSecurityGroupName", name + "nsg");
+         variables.put("networkSecurityGroupNameReference", "[resourceId('Microsoft.Network/networkSecurityGroups',variables('networkSecurityGroupName'))]");
+
+         List<NetworkSecurityRule> rules = new ArrayList<NetworkSecurityRule>();
+         for (int i = 0; i < ports.length; i++) {
+            NetworkSecurityRuleProperties ruleProperties = NetworkSecurityRuleProperties.builder()
+                    .description("default-allow-port-" + ports[i])
+                    .protocol(NetworkSecurityRuleProperties.Protocol.All)
+                    .access(NetworkSecurityRuleProperties.Access.Allow)
+                    .sourcePortRange("*")
+                    .destinationPortRange("*")
+                    .sourceAddressPrefix("*")
+                    .destinationAddressPrefix("*")
+                    .priority(1234 + i)
+                    .direction(NetworkSecurityRuleProperties.Direction.Inbound)
+                    .build();
+
+            NetworkSecurityRule networkSecurityRule = NetworkSecurityRule.create(
+                    "default-allow-port-" + ports[i],
+                    null,
+                    null,
+                    ruleProperties);
```

Comment:
Good question. I don't know why, but the fix should be straightforward .destinationPortRange(Integer.toString(ports[i]))